### ISSUE TITLE
Config MatrixTransform generation low precision

### ISSUE THREAD

**Issue by sobotka** [AcademySoftwareFoundation/OpenColorIO#577]
When using the MatrixTransform to generate a configuration line, the precision appears fixed at 6 decimal places. The configuration's YAML seems to be able to handle much more than this, including scientific notation if hand coded.

Is there a precision option I'm missing?

**Comment by KevinJW** [comment]
Relates to #389 

Kevin

**Comment by hodoulp** [comment]
I did some investigations on that topic to better understand the Matrix problem with the OCIO config file.

Here is some code to understand the float & double precisions using `std::numeric_limits<T>`:
```
#include <iostream>
#include <limits>

int main()
{
   std::cout << "float  => "
        << " max_digits10 = " << std::numeric_limits<float>::max_digits10 << ", "
        << " digits10 = " << std::numeric_limits<float>::digits10 << ", "
        << " digits = " << (2 + std::numeric_limits<float>::digits * 3010/10000)
        << std::endl;

   std::cout << "double => "
        << " max_digits10 = " << std::numeric_limits<double>::max_digits10 << ", "
        << " digits10 = " << std::numeric_limits<double>::digits10 << ", "
        << " digits = " << (2 + std::numeric_limits<double>::digits * 3010/10000)
        << std::endl;

    return 1;
}
```
The output is:
```
float  =>  max_digits10 = 9,  digits10 = 6,  digits = 9
double =>  max_digits10 = 17,  digits10 = 15,  digits = 17
```

When changing the code (i.e. OCIO & yaml following #389 changes) to use the `std::numeric_limits<float>::max_digits10`, the serialization/deserialization of floats have some rounding issues (i.e. various serialization/deserialization unit tests fail because of  string comparisons for float values: '0.11' versus '0.110000007'`).

So, I think that the good long-term solution is to have a Matrix of doubles (instead of floats) so double precision values will be saved/loaded by the current yaml version. However, the MatrixTransform public API would have to support doubles (and floats for backward compatibility).

**Note:**
The new Range Op only supports double values without any serialize/deserialize rounding issue (i.e. a Range unit test is validating some double values with 14 bit precision).

**Comment by hodoulp** [comment]
@sobotka Is the problem now fix in master following the merge of the pull requests #724 & #883?

**Comment by sobotka** [comment]
Will give it a roll this weekend. Thanks so much for pushing some changes @hodoulp !

**Comment by doug-walker** [comment]
This was fixed in PR #724.
